client/client.go

```diff
@@ -401,3 +424,26 @@ func (r *redirectedHTTPAction) HTTPRequest(ep url.URL) *http.Request {
 	orig.URL = &r.location
 	return orig
 }
+
+// randPermStartWithX returns, as a slice of n ints, a pseudo-random permutation
+// of the integers [0,n), which starts with x.
+func randPermStartWithX(r *rand.Rand, n, x int) []int {
+	p := r.Perm(n)
+	for i, k := range p {
```

Comment:
I will try this way. I prefer not to make pinned one twice because if it waits for timeout to fail, it may use much time.